### PR TITLE
Fix duplicate ajax requests

### PR DESCRIPTION
**Pull request by ausi**
Follow up to #35, #27

### CODE REVIEW

File: cq-prolyfill.js [hunk 1]
```diff
@@ -320,23 +319,29 @@ function preprocessSheet(sheet, callback) {
...
-
-			IN_FLIGHT[href].forEach(function(cb) { cb(CACHE[href]); });
-			delete IN_FLIGHT[href];
+			response = response || '';
+			requestCache[href].forEach(function(cachedCallback) {
+				setTimeout(function() {
+					cachedCallback(response);
+				});
+			});
+			requestCache[href] = response;
```
Comment (pl12133) [inline cq-prolyfill.js]: That's a smart way to combine the two cache objects into one.

File: cq-prolyfill.js [hunk 2]
```diff
@@ -317,10 +318,26 @@ function preprocessSheet(sheet, callback) {
  *                                    success or empty string on failure
  */
 function loadExternal(href, callback) {
+	var cacheEntryType = typeof requestCache[href];
+	if (cacheEntryType === 'string') {
+		callback(requestCache[href]);
+		return;
+	}
+	else if (cacheEntryType === 'object') {
```
Comment (developit) [inline cq-prolyfill.js]: Just double checking: will `requestCache[href]` ever be `null` (eg: failed xhr)?  If so, might be worth a truthiness check.

Comment (ausi) [inline cq-prolyfill.js]: `requestCache[href]` will be `undefined` initially, an array until the request is finished and a string afterwards. Failed XHRs are saved as an empty string, so it should never be `null`.

Comment (developit) [inline cq-prolyfill.js]: Perfect, that's what I figured but just wanted to make sure.

**Pull request merged**